src/builder.rs

```diff
@@ -0,0 +1,95 @@
+use std::path::{Path, PathBuf};
+
+use crate::Search;
+
+/// Builder for a [`Search`] instance, allowing for more complex searches
+pub struct SearchBuilder {
+    /// The location to search in, defaults to the current directory
+    search_location: PathBuf,
+    /// The search input, defaults to any word
+    search_input: Option<String>,
+    /// The file extension to search for, defaults to any file extension
+    file_ext: Option<String>,
+    /// The depth to search to, defaults to no limit
+    depth: Option<usize>,
+}
+
+impl SearchBuilder {
+    /// Build a new [`Search`] instance
+    #[allow(deprecated)]
```

Comment:
What does this line do ?

Reply:
Unshow the deprecated warning of `new` function

Reply:
Okay.